### PR TITLE
I-168 reorganize and document where messages come from

### PR DESCRIPTION
**Pull request by EzequielPostan**
Changes applied as discussed in #168 
- Classes `PowBlock` and `Mesages` should still be relocated.
- We need to review how to organize shared messages sent to more than one actor.

### CODE REVIEW

File: examples/src/main/scala/examples/hybrid/HLocalInterface.scala [hunk 1]
```diff
@@ -41,12 +40,12 @@ class HLocalInterface(override val viewHolderRef: ActorRef,
     if (!block) {
       mod match {
         case wb: PowBlock =>
-          posForgerRef ! StartForging
+          posForgerRef ! PosForger.ReceivableMessages.StartForging
```
Comment (ceilican) [inline examples/src/main/scala/examples/hybrid/HLocalInterface.scala]: What is the advantage of doing this change?

File: examples/src/main/scala/examples/hybrid/mining/PosForger.scala [hunk 2]
```diff
@@ -95,8 +106,8 @@ object PosForger extends ScorexLogging {
     HBoxStoredState,
     HWallet,
     SimpleBoxTransactionMemPool,
-    PosForgingInfo] = {
-    val f: CurrentView[HybridHistory, HBoxStoredState, HWallet, SimpleBoxTransactionMemPool] => PosForgingInfo = {
+    ReceivableMessages.PosForgingInfo] = {
```
Comment (ceilican) [inline examples/src/main/scala/examples/hybrid/mining/PosForger.scala]: maybe add a `import ReceivableMessages._` instead...

File: src/main/scala/scorex/core/NodeViewHolder.scala [hunk 3]
```diff
@@ -400,7 +400,18 @@ trait NodeViewHolder[P <: Proposition, TX <: Transaction[P], PMOD <: PersistentN
...
 
-  case class GetDataFromCurrentView[HIS, MS, VL, MP, A](f: CurrentView[HIS, MS, VL, MP] => A)
+  object ReceivableMessages extends LocallyGeneratedModifiersMessages with NodeViewHolderSharedMessages {
+    // Explicit request of NodeViewChange events of certain types.
+    case class GetNodeViewChanges(history: Boolean, state: Boolean, vault: Boolean, mempool: Boolean)
+    //a command to subscribe for events
+    case class Subscribe(events: Seq[EventType.Value])
+    case class GetDataFromCurrentView[HIS, MS, VL, MP, A](f: CurrentView[HIS, MS, VL, MP] => A)
+
+    // Moved fro NodeViewSynchronizer as this was only received here
```
Comment (ceilican) [inline src/main/scala/scorex/core/NodeViewHolder.scala]: i think this comment line could be removed.

@kushti, could you confirm?

File: src/main/scala/scorex/core/NodeViewHolderSharedMessages.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
+package scorex.core
+
+import scorex.core.NodeViewHolder.NodeViewHolderEvent
+import scorex.core.consensus.{HistoryReader, SyncInfo}
+import scorex.core.transaction.box.proposition.Proposition
+import scorex.core.transaction.{MempoolReader, Transaction}
+import scorex.core.transaction.state.StateReader
+
+trait NodeViewHolderSharedMessages {
```
Comment (ceilican) [inline src/main/scala/scorex/core/NodeViewHolderSharedMessages.scala]: shared with whom?

File: src/main/scala/scorex/core/network/NetworkControllerSharedMessages.scala [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package scorex.core.network
+
+import scorex.core.network.message.MessageSpec
+import scala.reflect.runtime.universe.TypeTag
+
+trait NetworkControllerSharedMessages {
```
Comment (ceilican) [inline src/main/scala/scorex/core/network/NetworkControllerSharedMessages.scala]: shared with whom?

File: src/main/scala/scorex/core/network/NodeViewSynchronizer.scala [hunk 6]
```diff
@@ -326,20 +328,24 @@ MR <: MempoolReader[TX]](networkControllerRef: ActorRef,
...
-                           modifierId: ModifierId)
+  object ReceivableMessages extends NodeViewHolderSharedMessages with NetworkControllerSharedMessages {
+    // getLocalSyncInfo messages
+    case object SendLocalSyncInfo
+    case class RequestFromLocal(source: ConnectedPeer, modifierTypeId: ModifierTypeId, modifierIds: Seq[ModifierId])
+    case class ResponseFromLocal[M <: NodeViewModifier](source: ConnectedPeer, modifierTypeId: ModifierTypeId, localObjects: Seq[M])
+    case class CheckDelivery(source: ConnectedPeer,
+                             modifierTypeId: ModifierTypeId,
+                             modifierId: ModifierId)
+    // Moves from NodeViewHolder as this is only received here
```
Comment (ceilican) [inline src/main/scala/scorex/core/network/NodeViewSynchronizer.scala]: i think this comment can be removed now.

File: src/main/scala/scorex/core/network/NodeViewSynchronizer.scala [hunk 7]
```diff
@@ -326,20 +328,24 @@ MR <: MempoolReader[TX]](networkControllerRef: ActorRef,
...
+    case class OtherNodeSyncingStatus[SI <: SyncInfo](remote: ConnectedPeer,
+                                                      status: History.HistoryComparisonResult.Value,
+                                                      remoteSyncInfo: SI,
+                                                      localSyncInfo: SI,
+                                                      extension: Option[Seq[(ModifierTypeId, ModifierId)]])
+    // Relocated from PeerManager as this events are only received here
```
Comment (ceilican) [inline src/main/scala/scorex/core/network/NodeViewSynchronizer.scala]: this too.

Thanks for these comments, though!

File: src/main/scala/scorex/core/network/peer/PeerManager.scala [hunk 8]
```diff
@@ -165,45 +168,35 @@ class PeerManager(settings: ScorexSettings, timeProvider: NetworkTimeProvider) e
 
 object PeerManager {
 
+  trait PeerManagerEvent
```
Comment (ceilican) [inline src/main/scala/scorex/core/network/peer/PeerManager.scala]: maybe this trait should be moved in the same place where `HandshakedPeer` and `DisconnectedPeer` were moved.

File: testkit/src/main/scala/scorex/testkit/properties/NodeViewSynchronizerTests.scala [hunk 9]
```diff
@@ -59,9 +58,10 @@ trait NodeViewSynchronizerTests[P <: Proposition,
     }
   }
 
-  import NodeViewHolder._   // NodeViewHolder's messages
-  import NodeViewSynchronizer._   // NodeViewSynchronizer's messages
-  import NetworkController._      // NetworkController's messages
+  import scorex.core.LocalInterface.ReceivableMessages.{BetterNeighbourAppeared, NoBetterNeighbour}
+  import NodeViewHolder.ReceivableMessages.{ModifiersFromRemote, CompareViews}  // NodeViewHolder's messages
```
Comment (ceilican) [inline testkit/src/main/scala/scorex/testkit/properties/NodeViewSynchronizerTests.scala]: we can now remove the comments from these lines, as they have become obvious.

File: src/main/scala/scorex/core/network/peer/PeerManager.scala (hunk 8 above)
Comment (EzequielPostan) [reply src/main/scala/scorex/core/network/peer/PeerManager.scala]: I did that at first, but it was also imported by the PeerManager and the statement 
`import NodeViewSynchronizer.ReceivableMessages.PeerManagerEvent`
looked strange, but I can relocate it.

File: testkit/src/main/scala/scorex/testkit/properties/NodeViewSynchronizerTests.scala (hunk 9 above)
Comment (EzequielPostan) [reply testkit/src/main/scala/scorex/testkit/properties/NodeViewSynchronizerTests.scala]: yep, indeed

File: src/main/scala/scorex/core/NodeViewHolderSharedMessages.scala (hunk 4 above)
Comment (EzequielPostan) [reply src/main/scala/scorex/core/NodeViewHolderSharedMessages.scala]: In the case of `NodeViewHolderSharedMessages`, it is shared between `NodeViewSynchronizer`, `LocalInterface` and `NodeViewHolder`

File: src/main/scala/scorex/core/network/NetworkControllerSharedMessages.scala (hunk 5 above)
Comment (EzequielPostan) [reply src/main/scala/scorex/core/network/NetworkControllerSharedMessages.scala]: In this case it is shared between  `NodeViewSynchronizer`, `PeerSynchronizer` and `NetworkController`

File: src/main/scala/scorex/core/NodeViewHolderSharedMessages.scala (hunk 4 above)
Comment (ceilican) [reply src/main/scala/scorex/core/NodeViewHolderSharedMessages.scala]: Maybe rename it to `NodeViewLocalInterfaceSharedMessages` then...

As discussed, these shared messages traits/objects are a temporary solution.

File: src/main/scala/scorex/core/network/peer/PeerManager.scala (hunk 8 above)
Comment (ceilican) [reply src/main/scala/scorex/core/network/peer/PeerManager.scala]: I see. Ok.
Could you just add a comment with a "todo" or "fixme", saying that we should reconsider the lcoation of this trait in the future.

**Pull request merged**